File: lib/java/test/java/appSecuritySpec.java
package app;

import org.junit.jupiter.api.DisplayName;
import org.junit.jupiter.api.Test;
import org.junit.jupiter.api.Tag;
import java.util.List;
import static org.junit.jupiter.api.Assertions.assertTrue;
import static org.junit.jupiter.api.Assertions.assertFalse;
import static org.junit.jupiter.api.Assertions.assertThrows;

@DisplayName("Security unit tests")
@Tag("security")
public class appSecuritySpec {

    @Test
    public void amount_bigger_than_IntMax_throw_exception() {
        assertThrows(ArithmeticException.class, () -> {
            Main app = new Main();
            boolean res = app.approval("2147483648");
        });
    }

    @Test
    public void amount_less_than_IntMin_throw_exception() {
        assertThrows(ArithmeticException.class, () -> {
            Main app = new Main();
            boolean res = app.approval("-2147483648");
        });
    }

    //@Test
    //public void amount_bigger_than_IntMax_after_surcharge_throw_exception() {
    //    assertThrows(ArithmeticException.class, () -> {
    //        Main app = new Main();
    //        //FIX ME
    //    });
    //}

    @Test
    public void amount_negative_throw_exception() {
        assertThrows(IllegalArgumentException.class, () -> {
            Main app = new Main();
            boolean res = app.approval("-1");
        });
    }

    @Test
    public void amount_zero_throw_exception() {
        assertThrows(IllegalArgumentException.class, () -> {
            Main app = new Main();
            boolean res = app.approval("0");
        });
    }

    @Test
    public void amount_null_throw_exception() {
        assertThrows(NullPointerException.class, () -> {
            Main app = new Main();
            boolean res = app.approval(null);
        });
    }
}

File: lib/scala/test/scala/app/appSecuritySpec.scala
package app

import org.scalatest.funsuite.AnyFunSuite
import org.scalatest.Tag

object Security extends Tag("app.tags.Security")

class appSecuritySpec extends AnyFunSuite {

  test("amount bigger than Int.MaxValue throws exception", Security) {
    assertThrows[NumberFormatException] {
      Main.approval("2147483648")
    }
  }

  test("amount less than Int.MinValue throws exception", Security) {
    assertThrows[NumberFormatException] {
      Main.approval("-2147483648")
    }
  }

  test("negative amount throws exception", Security) {
    assertThrows[IllegalArgumentException] {
      Main.approval("-1")
    }
  }

  test("zero amount throws exception", Security) {
    assertThrows[IllegalArgumentException] {
      Main.approval("0")
    }
  }

  test("null amount throws exception", Security) {
    assertThrows[NullPointerException] {
      Main.approval(null)
    }
  }
}
